packages/dd-trace/src/appsec/stack_trace.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const { calculateDDBasePath } = require('../util')
+
+const ddBasePath = calculateDDBasePath(__dirname)
+
+const LIBRARY_FRAMES_BUFFER = 20
+
+function getCallSiteList (maxDepth = 100) {
+  const previousPrepareStackTrace = Error.prepareStackTrace
+  const previousStackTraceLimit = Error.stackTraceLimit
+  let callsiteList
+  Error.stackTraceLimit = maxDepth
+
+  try {
+    Error.prepareStackTrace = function (_, callsites) {
+      callsiteList = callsites
+    }
+    const e = new Error()
+    e.stack
+  } finally {
+    Error.prepareStackTrace = previousPrepareStackTrace
+    Error.stackTraceLimit = previousStackTraceLimit
+  }
+
+  return callsiteList
+}
+
+function filterOutFramesFromLibrary (callSiteList) {
+  return callSiteList.filter(callSite => !callSite.getFileName().includes(ddBasePath))
+}
+
+function getFramesForMetaStruct (callSiteList, maxDepth = 32) {
+  const maxCallSite = maxDepth < 1 ? Infinity : maxDepth
```

Comment:
I don't think we need to chekc for maxDepth < 1 again here since we do it in reportStackTrace() no ?

Reply:
They are different.

The one checked in reportStackTrace is related to DD_APPSEC_MAX_STACK_TRACES - how many stack traces will be collected in each exploit prevention event.

The one highlighted in this comment, is related to DD_APPSEC_MAX_STACK_TRACE_DEPTH - how many frames of each stack trace will be collected.

Reply:
uuuh, `stackTraceLimit` and `maxDepth` are the same value, but you recheck for <0 in `getFramesForMetaStruct()`
![image](https://github.com/DataDog/dd-trace-js/assets/70517128/2e8f1451-8e13-4efc-9862-ec9f18b1f456)

Reply:
```javascript
    if (maxDepth < 1) maxDepth = Infinity
    const callSiteList = callSiteListGetter(maxDepth  + LIBRARY_FRAMES_BUFFER)
    const frames = getFramesForMetaStruct(callSiteList, maxDepth)
```

Reply:
Yes! Good catch!